pkg/scheduler/dependency_graph.go

```diff
@@ -336,29 +366,44 @@ func (rl *ReplicaList) Swap(i, j int) {
 	rl.Items[i], rl.Items[j] = rl.Items[j], rl.Items[i]
 }
 
-func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.DependencyGraphOptions) ([]string, error) {
-	label := labels.Set{"flow": flow.Name}
+func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.DependencyGraphOptions) ([]client.Replica, []client.Replica, error) {
+	name := flow.Name
+	if options.FlowInstanceName != "" {
+		name = options.FlowInstanceName
+	}
+	label := labels.Set{"flow": name}
 	var tmp ReplicaList
 	var maxCurrentTime time.Time
 
 	count := options.ReplicaCount
-	if !options.ReplicaCountIsRelative || options.ReplicaCount == 0 {
+	if options.FixedNumberOfReplicas || options.ReplicaCount <= 0 {
 		existingReplicas, err := sched.client.Replicas().List(api.ListOptions{
 			LabelSelector: labels.SelectorFromSet(label),
 		})
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		for _, item := range existingReplicas.Items {
 			if item.CreationTimestamp.After(maxCurrentTime) {
 				maxCurrentTime = item.CreationTimestamp.Time
 			}
 		}
 		tmp = ReplicaList(*existingReplicas)
-		if options.ReplicaCount == 0 {
-			count = len(tmp.Items)
+		if !options.FixedNumberOfReplicas && options.ReplicaCount <= 0 {
+			count = len(tmp.Items) + options.ReplicaCount
 		}
 	}
+	if count < options.MinReplicaCount && options.MinReplicaCount > 0 {
+		count = options.MinReplicaCount
+	}
+	if count > options.MaxReplicaCount && options.MaxReplicaCount > options.MinReplicaCount && options.MaxReplicaCount > 0 {
```

Comment:
MaxReplicaCount = 0 means there is no maximum since it is the default value in golang. There is no practical sense to have maximum set to 0 as such flow couldn't be created at all